spec/factories/test_batch.rb

```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :test_batch do
+    association :course, factory: :course
```

Comment:
Look at how other model factories define a `course` association; they use a default course, which you should do here. This avoids needing to include the `before(:create)` block in the `test_runs.rb` factory you added.